app/src/main/java/com/isc/hermes/controller/IncidentFormController.java

```diff
@@ -105,19 +126,76 @@ private void handleUploadResponse(Integer responseCode) {
+                incidentTypeIcons.length == incidentTypes.length) {
+            for (int i = 0; i < incidentTypes.length; i++) {
+                IncidentTypeButton incidentTypeButton = new IncidentTypeButton();
+                Button button = incidentTypeButton.getIncidentTypeButton(
+                        context,
+                        incidentTypes[i].toLowerCase(),
+                        Color.parseColor((String) incidentTypeColors[i]),
+                        incidentTypeIcons[i]);
+                setIncidentButtonAction(button);
+
+                incidentTypesContainer.addView(button);
+            }
+        } else {
+            Timber.i(String.valueOf(R.string.array_size_text_timber));
+        }
+
+        setEstimatedTimePicker();
+    }
 
+    /**
+     * This method set the content of the time picker,
+     * and set the number picker according the kind of time.
+     */
+    private void setEstimatedTimePicker() {
         Spinner incidentEstimatedTime = ((AppCompatActivity) context).findViewById(R.id.incident_time_spinner);
         ArrayAdapter<CharSequence> adapterTime=ArrayAdapter.createFromResource(context, R.array.incidents_estimated_time, R.layout.incident_spinner_items);
         adapterTime.setDropDownViewResource(android.R.layout.simple_spinner_item);
-        incidentEstimatedTime.setAdapter(adapterTime);
 
+        incidentEstimatedTime.setAdapter(adapterTime);
         NumberPicker incidentTimePicker = ((AppCompatActivity) context).findViewById(R.id.numberPicker);
-        incidentTimePicker.setMinValue(1);
-        incidentTimePicker.setMaxValue(10);
+        incidentTimePicker.setDescendantFocusability(NumberPicker.FOCUS_BLOCK_DESCENDANTS);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            incidentTimePicker.setTextColor(Color.BLACK);
+        }
+        incidentEstimatedTime.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                String selectedOption = adapterTime.getItem(position).toString();
+
+                switch (selectedOption) {
+                    case "min" -> {
+                        incidentTimePicker.setMinValue(1);
+                        incidentTimePicker.setMaxValue(60);
+                    }
+                    case "hr" -> {
+                        incidentTimePicker.setMinValue(1);
+                        incidentTimePicker.setMaxValue(24);
+                    }
+                    case "day" -> {
+                        incidentTimePicker.setMinValue(1);
+                        incidentTimePicker.setMaxValue(31);
+                    }
+                    case "month" -> {
+                        incidentTimePicker.setMinValue(1);
+                        incidentTimePicker.setMaxValue(12);
+                    }
+                }
```

Comment:
This method is too long, could you move the logic to a separate method?

Reply:
Done